src/app/modules/shared_modules/keycard_popup/internal/manage_keycard_accounts_state.nim

```diff
@@ -15,6 +15,21 @@ method getNextPrimaryState*(self: ManageKeycardAccountsState, controller: Contro
     return createState(StateType.CreatingAccountNewSeedPhrase, self.flowType, nil)
   if self.flowType == FlowType.SetupNewKeycardOldSeedPhrase:
     return createState(StateType.CreatingAccountOldSeedPhrase, self.flowType, nil)
+  if self.flowType == FlowType.ImportFromKeycard:
+    let numOfProcessedAccounts = controller.getKeyPairForProcessing().getAccountsModel().getCount()
+    let totalNumOfAccountsToBeProcessed = controller.getKeyPairHelper().getAccountsModel().getCount()
+    if numOfProcessedAccounts < totalNumOfAccountsToBeProcessed:
+      let accountItem = controller.getKeyPairHelper().getAccountsModel().getItemAtIndex(numOfProcessedAccounts) # numOfProcessedAccounts is index of next acc which need to be processed
+      if accountItem.isNil:
+        # should never be here
```

Comment:
show we log an error here?

Reply:
@dlipicar I added the comment for future maintaining only that we should never get there, actually based on the logic, flow it's impossible. Also for the correctness purpose in terms of coding the code which will display failor state to a user is already there, because of that no need for any additional logging.